pkg/generator_test.go

```diff
@@ -760,6 +760,32 @@ func (s *GeneratorSuite) TestReplaceTypePackageMultiple() {
 	})
 }
 
+func (s *GeneratorSuite) TestReplaceTypeGeneric() {
+	cfg := GeneratorConfig{InPackage: false, ReplaceType: []string{
+		"github.com/vektra/mockery/v2/pkg/fixtures.ReplaceGeneric[-TImport]=github.com/vektra/mockery/v2/pkg/fixtures/redefined_type_b.B",
+		"github.com/vektra/mockery/v2/pkg/fixtures.ReplaceGeneric[TConstraint]=github.com/vektra/mockery/v2/pkg/fixtures/constraints.Integer",
+		"github.com/vektra/mockery/v2/pkg/fixtures.ReplaceGenericSelf[-T]=github.com/vektra/mockery/v2/pkg/fixtures.*ReplaceGenericSelf",
+	}}
+
+	s.checkGenerationRegexWithConfig("generic.go", "ReplaceGeneric", cfg, []regexpExpected{
+		// type ReplaceGeneric[TConstraint constraints.Integer, TKeep interface{}] struct
+		{true, regexp.MustCompile(`type ReplaceGeneric\[TConstraint constraints.Integer\, TKeep interface\{\}] struct`)},
+		// func (_m *ReplaceGeneric[TConstraint, TKeep]) A(t1 test.B) TKeep
+		{true, regexp.MustCompile(`func \(_m \*ReplaceGeneric\[TConstraint, TKeep\]\) A\(t1 test\.B\) TKeep`)},
+		// func (_m *ReplaceGeneric[TConstraint, TKeep]) B() test.B
+		{true, regexp.MustCompile(`func \(_m \*ReplaceGeneric\[TConstraint, TKeep\]\) B\(\) test\.B`)},
+		// func (_m *ReplaceGeneric[TConstraint, TKeep]) C() TConstraint
+		{true, regexp.MustCompile(`func \(_m \*ReplaceGeneric\[TConstraint, TKeep\]\) C\(\) TConstraint`)},
+	})
+
+	s.checkGenerationRegexWithConfig("generic.go", "ReplaceGenericSelf", cfg, []regexpExpected{
+		// type ReplaceGenericSelf struct
+		{true, regexp.MustCompile(`type ReplaceGenericSelf struct`)},
+		// func (_m *ReplaceGenericSelf) A() *ReplaceGenericSelf
+		{true, regexp.MustCompile(`func \(_m \*ReplaceGenericSelf\) A\(\) \*ReplaceGenericSelf`)},
+	})
+}
+
```

Comment:
I do not like these kinds of tests and am asking people not to do them. Any changes to the formatting can/will cause these kinds of tests to fail. Instead I prefer if we just create a new test file, probably in `pkg/fixtures/test` that checks these various scenarios through a functional test, rather than string comparison. You can just rely on the compiler to fail if the param types are mismatched, or maybe use type assertion through an empty interface in the test (actually, type assertion might be better in this case so that we don't cause all the tests to fail if the mock implementation is wrong somehow).

Reply:
Good idea! I've added some better test cases and removed these. I didn't quite figure out exactly how we could test just with type assertions, at the moment the compiler would complain if the types no longer match